OPEN_PROBLEMS/MUTABLE_DATA.md

```diff
@@ -0,0 +1,135 @@
+# Mutable Data (Naming, Real-Time, Guarantees)
+
+## Description
+
+Mutable Data in a Permanent Datastore? WAT?! If this is your reaction, don't despair, we've been there too. However, the good news is that achieving mutability over immutable datastores is not a new idea, in fact, if you are checking this document through Git/GitHub, you are experiencing just that.
+
+The simple trick is to use pointers. While each data struct will have its own reference (in the case of IPFS, they will have a CID), mutability is achieved by updating the pointer to point to a new version of that data structure, just like when you update your master branch (pointer) to your latest commit (immutable reference) on your Git projects.
+
+Having Mutable Data is essential to create a whole new set of applications for the Web 3.0. These applications will have different types of requirements, from millions of records being update every second, to millions of users interacting and mutating a data source.
+
+Creating a sound solution to achieve Mutable Data for the dWeb requires considering the three parts of the problem:
+
+- **Update Propagation** - As noted, diffent applications will have different requirementsm, these can go from real-time delivery, to delivery at least/most once, confirmation of delivery, conflict resolution, etc.
+- **Update Authentication** - How to trust the parties that are propagating the updates, how to trust that they are not hiding any important update
+- **Interface/API** - This is where things that shape up to feel like a database start to show. From convinence features to powerful APIs that enable a user to query, update and search over multiple records
+
+All of these problems should be solved to work on a Distributed Network scenario, this means: disconnected, offline first and over unreliable networks.
+
+## State of the Art
+
+> This survey on the State of the Art is not by any means complete, however, it should provide a good entry point to learn what are the existing work. If you have something that is fundamentally missing, please consider submitting a PR to augment this survey. 
+
+### Within the IPFS Ecosystem
+> Existing attempts and strategies
+
+##### IPNS
```

Comment:
@hugomrdias @aschmahmann where can I find your latest slides, talks and notes on IPNS with multiple routers?

Reply:
- Talk https://hackmd.io/@hugodias/ipns#/

Reply:
Also include:
- [Improve IPNS spec]https://github.com/ipfs/specs/issues/205)
- [IPNS-over-PubSub as an Independent Transport](https://github.com/ipfs/specs/pull/218)

Reply:
There's also this issue: ipfs/specs#198